pkg/controller/service/servicecontroller.go

```diff
@@ -314,6 +319,14 @@ func (s *ServiceController) createLoadBalancerIfNeeded(key string, service *v1.S
 
 func (s *ServiceController) persistUpdate(service *v1.Service) error {
 	var err error
+
+	// Make a copy so we don't mutate the shared informer cache
+	copy, err := api.Scheme.DeepCopy(service)
```

Comment:
But this controller can potentially update `service.Status.LoadBalancer`, so I assume we need a deep copy prior to that happening.